src/server/util/authenticationHelpers.js

```diff
@@ -14,6 +14,15 @@ const getUserInDatabase = async function(username, email) {
     return result.rows;
 };
 
+const getUserById = async function(id) {
```

Comment:
@gnarf @ktbee How does the consumer of the roles API know where to get the user id? Or should we assume that the consumer just has it?

Reply:
Good Q, previously I got it from an email address or username, but in these edits we'd get it from the API request (ie `req.params.user_id`). I guess we will likely need another endpoint to get the user id and then hit it on the front end before we make our request to the roles API? Do I have that right @gnarf?

Reply:
If you are "editing" or "viewing" roles, you definitely have the user you want to edit's id... You are probably on the users edit page doing this, etc... Making this endpoint also search by username/email causes the weird "what if we find both" situations, which will easily drift to be different resollutions on this endpoint vs other endpoints (perhaps one searches username first, but the other doesn't).